### PR TITLE
Add Server API to get tenant pools

### PR DESCRIPTION
**Pull request by MeihanLi**
Add a new Server API GET /instance/pools in InstanceResource to get server pools. `feature` 

It returns an empty map if the server is not assigned to any pool number

Tested locally:
1. server is not assigned to any pool number
```
❯ curl localhost:7500/instance/pools | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     2  100     2    0     0    153      0 --:--:-- --:--:-- --:--:--   333
{}
```
2. server is assigned to pool numbers                                                                                                                                                                              
```
❯ curl localhost:7500/instance/pools | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    58  100    58    0     0   5070      0 --:--:-- --:--:-- --:--:-- 11600
{
  "DefaultTenant_OFFLINE": "0",
  "DefaultTenant_REALTIME": "0"
}
```

### CODE REVIEW

**Comment by MeihanLi** [comment]
cc: @chenboat can you help review it? Thanks!

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/InstanceResource.java [hunk 1]
```diff
@@ -73,4 +75,17 @@ public List<String> getInstanceTags() {
...
+
+  @GET
+  @Path("pools")
+  @ApiOperation(value = "Tenant pools for current instance")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")
+  })
+  @Produces(MediaType.APPLICATION_JSON)
+  public Map<String, String> getInstancePools() {
+    InstanceConfig instanceConfig = HelixHelper.getInstanceConfig(_helixManager, _instanceId);
```
Comment (ankitsultana) [inline pinot-server/src/main/java/org/apache/pinot/server/api/resources/InstanceResource.java:87]: need null check for instanceConfig (similar to getInstanceTags)

Comment (MeihanLi) [reply pinot-server/src/main/java/org/apache/pinot/server/api/resources/InstanceResource.java:87]: Thanks Ankit, addressed

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/InstanceResource.java [hunk 2]
```diff
@@ -73,4 +75,20 @@ public List<String> getInstanceTags() {
     }
     return Collections.emptyList();
   }
+
+  @GET
+  @Path("pools")
+  @ApiOperation(value = "Tenant pools for current instance")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")
+  })
```
Comment (chenboat) [inline pinot-server/src/main/java/org/apache/pinot/server/api/resources/InstanceResource.java:84]: 1) can you add the javadoc to explain return result types? i.e.,The summary in the PR looks good.
2) Can you add a Pinot doc reference to the definition of pool in Pinot?

**Review by chenboat — APPROVED** [review]

**Review by ankitsultana — APPROVED** [review]

**Pull request merged**